### PR TITLE
Checkout Swift LLVM Bindings repo

### PR DESCRIPTION
**Pull request by egorzhdan**
This is similar to https://github.com/apple/swift/pull/60193.

### CODE REVIEW

**Comment by egorzhdan** [comment]
@swift-ci please test

**Comment by justice-adams-apple** [comment]
Hi @egorzhdan , this will likely break things on any branch that isn't `main` because there won't be a repository defined for that specific branch. Essentially a key error will occur in python.

For example, this is how our CI is run on 5.7:

```
./run release/5.7 --assertions --build-config=debug

** RUN **
Traceback (most recent call last):
  File "swift-source-compat-suite/./run", line 325, in <module>
    sys.exit(main())
  File "swift-source-compat-suite/./run", line 35, in main
    common.clone_repos()
  File "swift-source-compat-suite/common.py", line 188, in clone_repos
    branches[swift_branch]['swift-llvm-bindings'], workspace
KeyError: 'swift-llvm-bindings'
```

Is your intention to only clone this repo when building/testing the main branch?

File: common.py [hunk 1]
```diff
@@ -182,6 +183,10 @@ def clone_repos():
         '{}/llvm-project '.format(
             branches[swift_branch]['llvm-project'], workspace
         ),
+        '{} git@github.com:apple/swift-llvm-bindings.git'
```
Comment (justice-adams-apple) [inline common.py:186]: You will need to add a space here so the git clone command doesn't thing the whole string is the repository name

```suggestion
        '{} git@github.com:apple/swift-llvm-bindings.git '
```

**Comment by justice-adams-apple** [comment]
One thing to note is that you are in a particular position where your changes will not be tested by the PR CI jobs because we run with `--skip-clone` when invoking `run.py`

We defer the dependency cloning to the build-script itself

**Comment by egorzhdan** [comment]
Thanks @justice-adams-apple!

> Is your intention to only clone this repo when building/testing the main branch?

Yeap, this is correct. This repo will only be used with `main`, and with release branches that will be created in the future.

> One thing to note is that you are in a particular position where your changes will not be tested by the PR CI jobs because we run with --skip-clone when invoking run.py

Oh interesting, I didn't realize that. I'll make sure to test this locally.

I was mostly trying to mirror another patch that added the "swift-experimental-string-processing" repo (https://github.com/apple/swift-source-compat-suite/pull/635), since our checkout scheme is quite similar, but I'll double check that this doesn't produce errors for existing release branches.

File: common.py (hunk 1 above)
Comment (egorzhdan) [reply common.py]: Fixed, thanks for noticing!

**Comment by justice-adams-apple** [comment]
@egorzhdan If you needed this in quickly you could likely just add `swift-llvm-bindings': 'stable/20211026',` to each of the repo->branch maps like you did with `main`.

However this will just mean that the release branches which don't need to actually clone that repository will do so anyway. It's a quick way to get this in without breaking things, but adds some small overhead of cloning a project unnecessarily. 

The proper way would probably be to update the `clone_repos()` method of `common.py` with some logic to figure out which branch needs which set of projects based off the dictionary in question

Happy to help out either way, just let me know

**Comment by egorzhdan** [comment]
@justice-adams-apple I tweaked the `clone_repos()` logic slightly to make sure it doesn't try to checkout `swift-llvm-bindings` for the existing release branches. I'll run this locally to make sure it doesn't break something else. Could you please take another look at this patch?

**Comment by egorzhdan** [comment]
@swift-ci please test

**Review by justice-adams-apple — APPROVED** [review]
LGTM. I think this may become a bit complex if future releases all have a varying degree of repositories, but I'll address that when it happens.

I think the simplicity of this works for now. Thanks for the fix!

**Comment by egorzhdan** [comment]
Thanks @justice-adams-apple!

I updated the branch name after rebranch.

**Comment by egorzhdan** [comment]
@swift-ci please test

**Pull request merged**